generator/tag_admixer.hpp

```diff
@@ -147,66 +147,85 @@ class TagAdmixer
 class TagReplacer
 {
 public:
+  using Replacements = std::map<OsmElement::Tag, std::vector<OsmElement::Tag>>;
+
   TagReplacer() = default;
 
   explicit TagReplacer(std::string const & filePath)
   {
     std::ifstream stream(filePath);
 
-    OsmElement::Tag tag;
-    std::vector<std::string> values;
     std::string line;
     while (std::getline(stream, line))
     {
-      if (line.empty())
+      strings::Trim(line);
+      if (line.empty() || line[0] == '#')
         continue;
 
-      strings::SimpleTokenizer iter(line, " \t=,:");
-      if (!iter)
-        continue;
-      tag.m_key = *iter;
-      ++iter;
-      if (!iter)
-        continue;
-      tag.m_value = *iter;
+      auto keyIt = line.find("=");
+      CHECK(keyIt != std::string::npos, ("Cannot find source tag key in", line));
+      auto key = line.substr(0, keyIt);
+      strings::Trim(key);
+
+      // Skip '='.
+      ++keyIt;
+      auto valueIt = line.find(" : ", keyIt);
+      CHECK(valueIt != std::string::npos, ("Cannot find source tag value in", line));
+      auto value = line.substr(keyIt, valueIt - keyIt);
+      strings::Trim(value);
+
+      // Skip ' : '.
+      valueIt += 3;
 
-      values.clear();
-      while (++iter)
-        values.push_back(*iter);
+      auto rawReplacements = line.substr(valueIt);
+      strings::Trim(rawReplacements);
+      CHECK(!rawReplacements.empty(), ("Empty replacement in", line));
 
-      if (values.size() >= 2 && values.size() % 2 == 0)
-        m_entries[tag].swap(values);
+      auto const replacements = strings::Tokenize(rawReplacements, ",");
+      for (auto const & replacement : replacements)
+      {
+        auto kv = strings::Tokenize(replacement, "=");
+        CHECK_EQUAL(kv.size(), 2, ("Cannot parse replacement tag:", replacement, "in line:", line));
```

Comment:
nit: print the line number instead of the whole line

Reply:
done